### PR TITLE
feat: adding music player project for new JS curriculum

### PR DESCRIPTION
**Pull request by jdwilkin4**
This is currently the third project in the new JS curriculum.  Since we are starting to introduce more advanced JS concepts like map, I think it would be best to start providing the HTML and CSS for them. They will have two projects before this to practice building out the HTML and CSS before getting to the JS.

Link to live project 
https://music-playlist-fcc.netlify.app/

## Concepts covered

- [x] first introduction of `getElementById()`
- [x] review of how to access array elements
- [x] review of how to access object properties using dot notation
- [x] `Audio()` constructor
- [x] template literals and interpolation
- [x] first intro of `map` array method
- [x] first intro of default parameters
- [x] first use of `innerHTML` I believe 
- [x] review of ternary operator
- [x] first use of the not operator I believe 
- [x] review of `length` property
- [x] review of `Math.floor` and `Math.random`
- [x] first intro to array destructing to swap variables
- [x] first mention of `querySelectorAll` I believe
- [x] first mention of `confirm`web api
- [x] first mention of `remove` web api
- [x] first mention of `pause` web api
- [x] first mention of `addEventListener`
- [x] first intro to `forEach` method
- [x] first intro to `findIndex`
- [x] first intro to while loop I believe
- [x] first mention of `splice` method

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

### CODE REVIEW

**Comment by scissorsneedfoodtoo** [comment]
Hey @jdwilkin4, just took a look and everything is looking good. I also went through your helpful checklist and updated the project-based curriculum spreadsheet based on that.

Everything is looking good, I just have a couple of minor suggestions regarding the mp3 filenames, which I'll leave as review comments.

File: music-playlist-app/index.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+const previousBtn = document.getElementById("previous-btn");
+const nextBtn = document.getElementById("next-btn");
+const shuffleBtn = document.getElementById("shuffle-btn");
+let isPlaying = false;
+let currentSong = 0;
+
+// This could be a nice review for objects nested inside arrays and how to access those properties
+let playlist = [
+  {
+    id: "We-Are-Going-to-Make-it",
```
Comment (scissorsneedfoodtoo) [inline music-playlist-app/index.js:15]: Filename / id suggestion:

```suggestion
    id: "we-are-going-to-make-it",
```

File: music-playlist-app/index.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    audio: "audio/We-Are-Going-to-Make-it.mp3",
+    title: "We Are Going to Make it",
+    artist: "Quincy Larson",
+  },
+  {
+    id: "Cant-Stay-Down",
```
Comment (scissorsneedfoodtoo) [inline music-playlist-app/index.js:21]: Filename / id suggestion:

```suggestion
    id: "cant-stay-down",
```

File: music-playlist-app/index.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    audio: "audio/Can't-Stay-Down.mp3",
+    title: "Can't Stay Down",
+    artist: "Quincy Larson",
+  },
+  {
+    id: "Collider-in-My-Head",
```
Comment (scissorsneedfoodtoo) [inline music-playlist-app/index.js:27]: Filename / id suggestion:

```suggestion
    id: "collider-in-my-head",
```

File: music-playlist-app/index.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    audio: "audio/Collider-in-My-Head.mp3",
+    title: "Collider in My Head",
+    artist: "Quincy Larson",
+  },
+  {
+    id: "Cruising-for-a-Musing",
```
Comment (scissorsneedfoodtoo) [inline music-playlist-app/index.js:33]: Filename / id suggestion:

```suggestion
    id: "cruising-for-a-musing",
```

File: music-playlist-app/index.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    audio: "audio/Cruising-for-a-Musing.mp3",
+    title: "Cruising for a Musing",
+    artist: "Quincy Larson",
+  },
+  {
+    id: "The-Surest-Way-Out-is-Through",
```
Comment (scissorsneedfoodtoo) [inline music-playlist-app/index.js:39]: Filename / id suggestion:

```suggestion
    id: "the-surest-way-out-is-through",
```

**Comment by jdwilkin4** [comment]
I have updated the id file names and created a PR to add the audio files to the `cdn` repo
https://github.com/freeCodeCamp/cdn/pull/149

I will go ahead and merge this since it is good to go now.

**Comment by bbsmooth** [comment]
@jdwilkin4 Do you want any accessibility feedback on this?

**Comment by jdwilkin4** [comment]
@bbsmooth Sure 👍

**Comment by bbsmooth** [comment]
@jdwilkin4 I guess I should have asked this the first time. What would be the best way to give you the feedback? Opening issues in this repo?

**Comment by jdwilkin4** [comment]
Opening an issue works for me. 
If you tag me in an issue, then I can apply those changes and open a new PR 👍 
Thanks! 😄

**Pull request merged**